Add new hook: actionModifyHtmlPurifierConfig for make the TinyMCEEditor more flexible and extensible via module

Codencode opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 
Please take the time to edit the "Answers" rows below with the necessary information.
Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | new hook for Make the TinyMCEEditor more flexible and extensible via module
| Type?             | improvement
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Idea https://github.com/PrestaShop/PrestaShop/discussions/39259
Related PRs | https://github.com/PrestaShop/PrestaShop/pull/39277
| Sponsor company   | @Codencode 
| How to test?      |

Codencode committed:
Add new hook: actionModifyHtmlPurifierConfig

Quetzacoalt91 commented on upgrade/sql/8.2.2.sql:
```diff
@@ -19,7 +19,9 @@ INSERT INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`
-  (NULL, 'actionOrderHasBeenDelivered', 'Called when checking if an order has been delivered', 'Allows modules to override or react to the hasBeenDelivered() method of the Order class.', '1')
+  (NULL, 'actionOrderHasBeenDelivered', 'Called when checking if an order has been delivered', 'Allows modules to override or react to the hasBeenDelivered() method of the Order class.', '1'),
+  -- https://github.com/PrestaShop/PrestaShop/pull/39261
+  (NULL, 'actionModifyHtmlPurifierConfig', 'Called when configuring HTMLPurifier', 'Allows modules to modify the HTMLPurifier definition by adding custom allowed HTML elements or attributes during Tools::purifyHTML().', '1')
```
PrestaShop 8.2.2 is already published, so the new hook can't go in this file.

Let's wait for the review from the Core team first, they will let you know where it should go but it may be for PS 9.0.1

Codencode replied:
@Quetzacoalt91
You're right! Okay, we'll wait.
So I'll probably need to switch branches for the PR https://github.com/PrestaShop/PrestaShop/pull/39261, right?

Quetzacoalt91 replied:
Likely, yes

Codencode replied:
Ok, however, in my opinion, it could be useful to already have it in version 8—it's essentially a hook that would make extending the editor easier via a module without any overrides.

Quetzacoalt91 replied:
I understand your point. I just wanted to give an insight about what may happen while the other PR is reviewed.

Codencode replied:
Yeah, yeah, of course

Codencode commented:
@Quetzacoalt91,
I've created a new PR targeting the 9.0.x branch, so I removed the instruction from the 8.2.2.sql file and added it to the 9.0.1.sql file.

Quetzacoalt91 commented on upgrade/sql/9.0.1.sql:
```diff
@@ -0,0 +1,7 @@
+INSERT INTO `PREFIX_hook` (`id_hook`, `name`, `title`, `description`, `position`) VALUES
+  -- https://github.com/PrestaShop/PrestaShop/pull/39277
+  (NULL, 'actionModifyHtmlPurifierConfig', 'Called when configuring HTMLPurifier', 'Allows modules to modify the HTMLPurifier definition by adding custom allowed HTML elements or attributes during Tools::purifyHTML().', '1')
+ON DUPLICATE KEY UPDATE `title` = VALUES(`title`), `description` = VALUES(`description`);
```
```suggestion
ON DUPLICATE KEY UPDATE `title` = VALUES(`title`), `description` = VALUES(`description`);

```

Let's add one new empty line and we're good.

Codencode replied:
@Quetzacoalt91
Done

Thank you!

Codencode commented:
Hi @Quetzacoalt91, is this PR still blocked because something is missing?
Also, since I created the backport for version 8.2.3 https://github.com/PrestaShop/PrestaShop/pull/39432 (in case it needs to be released), should I add the instructions directly here, or should I create a new dedicated PR?
Thanks 🙏

Quetzacoalt91 commented:
@Codencode Can you please rebase your PR? We had an issue with PHPStan that prevented the CI to succeed which has been fixed since you opened the pull-request.

This PR goes with https://github.com/PrestaShop/PrestaShop/pull/39277. With the backport on the branch 8.2.x, note that we will need another PR on Update Assistant if https://github.com/PrestaShop/PrestaShop/pull/39432 is merged as we will need the hook to be added on 8.2.3 as well.

Codencode commented:
@Quetzacoalt91,

If I understand correctly, I need to:

Perform a rebase to add the subsequent commits to this PR

Create a new PR for ISSUE PrestaShop/PrestaShop#39432, which is the backport for version 8.2.x

Is that right?

Thanks a lot!

Quetzacoalt91 commented:
Exactly :)

Codencode commented:
@Quetzacoalt91,

I performed the rebase, but some tests are failing. Did I forget to do something?

This one is the PR for version 8.2.3: https://github.com/PrestaShop/autoupgrade/pull/1447

Quetzacoalt91 commented:
This is likely unrelated to your changes. I think the UI tests need a fix.

AureRita reviewed (approved):
Hi @Codencode 

Thank you for your PR, I tested it and it seems to works as you can see : 

https://github.com/user-attachments/assets/a3e7582e-fc65-44fa-9b5f-d806e9b046de


Tested from : 
8.1.5 to 9.1.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you
